pupil-spa/src/app/services/login-error-diagnostics/login-error-diagnostics.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { LoginErrorService } from '../login-error/login-error.service';
+import { APP_CONFIG } from '../config/config.service';
+import { HttpClient } from '@angular/common/http';
+import { first } from 'rxjs/operators';
+import { WindowRefService} from '../window-ref/window-ref.service';
+
+@Injectable()
+export class LoginErrorDiagnosticsService {
+
+  protected window: any;
+
+  constructor(
+    private http: HttpClient,
+    private loginErrorService: LoginErrorService,
+    private windowRefService: WindowRefService
+  ) {
+    this.window = windowRefService.nativeWindow;
+  }
+
+  async process(err) {
+    if (err.status !== 0) {
+      return;
+    }
+    const canAccessAuthURL = await this.canAccessURL(APP_CONFIG.authURL);
+    if (!canAccessAuthURL) {
+      this.loginErrorService.changeMessage(`Connection refused to ${APP_CONFIG.authURL}`);
+    }
+    const hasActiveInternetConnection = this.window.navigator.onLine;
+    if (!hasActiveInternetConnection) {
+      this.loginErrorService.changeMessage('Internet disconnected');
```

Comment:
It looks like the results of the URL check could be overwritten by the status of `window.navigator.onLine`.  If the URL check was successful then this makes no sense as it would be browser error.

Reply:
@jon-shipley updated based on discussion